gnovm/pkg/gnolang/preprocess.go

```diff
@@ -2125,8 +2130,19 @@ func preprocess1(store Store, ctx BlockNode, n Node) Node {
 
 			// TRANS_LEAVE -----------------------
 			case *BranchStmt:
+
+				notAllowedFunc := func(s string) {
+					_, isFunc := last.(*FuncLitExpr)
+
+					if isFunc {
+						panic(fmt.Sprintf("%s statement out of place", s))
+					}
+				}
+
 				switch n.Op {
 				case BREAK:
+					notAllowedFunc("break")
```

Comment:
Your fix seems incomplete. Consider the following case where the bug is not catched:

```go
package main

func main() {
    for i := 0; i < 10; i++ {
        if i == 1 {
            _ = func() int {
                if true {
                    break
                }
                return 11
            }()
        }
        println(i)
    }
    println("wat???")
}

// Error:
// main/files/for24.gno:8:17: break statement out of place
```